### ISSUE TITLE
lirios-wayland fails to compile with QtWayland 5.12

### ISSUE THREAD

**Issue by n3rdopolis** [liri-archive/wayland-archived#46]
I thought it was an upstream issue, as a QtWayland header is including a non-existing header that looks like it should be in QtWayland, but https://bugreports.qt.io/browse/QTBUG-71452 was closed

The error I get is
```
compiling fullscreenshellsurface.cpp
ERROR: /usr/bin/g++ -O2 -Wall -Wextra -m32 -pipe -fexceptions -fvisibility=hidden -fvisibility-inlines-hidden -fPIC -DNDEBUG -DQT_CORE_LIB -DQT_NO_DEBUG -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_SERVICE_SUPPORT_LIB -DQT_WAYLANDCLIENT_LIB -I/srcbuild/lirios-wayland/src/plugins/shellintegrations/fullscreen-shell -I/srcbuild/lirios-wayland/default/fullscreen-shell.0dc7dd67 -I/opt/include -I/opt/include/QtCore -I/opt/mkspecs/linux-g++ -I/srcbuild/lirios-wayland/default/fullscreen-shell.0dc7dd67/qt.headers -I/opt/include/QtCore/5.12.0 -I/opt/include/QtCore/5.12.0/QtCore -I/opt/include/QtDBus -I/opt/include/QtGui -I/opt/include/QtGui/5.12.0 -I/opt/include/QtGui/5.12.0/QtGui -I/opt/include/QtServiceSupport -I/opt/include/QtServiceSupport/5.12.0 -I/opt/include/QtServiceSupport/5.12.0/QtServiceSupport -I/opt/include/QtWaylandClient -I/opt/include/QtWaylandClient/5.12.0 -I/opt/include/QtWaylandClient/5.12.0/QtWaylandClient -std=c++11 -o /srcbuild/lirios-wayland/default/fullscreen-shell.0dc7dd67/3a52ce780950d4d9/fullscreenshellsurface.cpp.o -c /srcbuild/lirios-wayland/src/plugins/shellintegrations/fullscreen-shell/fullscreenshellsurface.cpp

ERROR: In file included from /srcbuild/lirios-wayland/src/plugins/shellintegrations/fullscreen-shell/fullscreenshellsurface.cpp:24:
/opt/include/QtWaylandClient/5.12.0/QtWaylandClient/private/qwaylandscreen_p.h:58:10: fatal error: QtWaylandClient/private/qwayland-xdg-output-unstable-v1.h: No such file or directory
 #include <QtWaylandClient/private/qwayland-xdg-output-unstable-v1.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ERROR: Process failed with exit code 1.
The following products could not be built for configuration default:
eglfs-kms-integration, fullscreen-shell
```

Not sure how to generate the QtWaylandClient/private/qwayland-xdg-output-unstable-v1.h header?

**Comment by plfiorini** [comment]
Ah looks like the same problem hitting the decoration.
I'm starting to be pissed off by all those private header issues, I don't want to touch them with a 10 foot pole any more than I really need to.

This summer I spoke to Johan and he was positive on getting the full screen shell plugin upstream, it's just getting stuck somewhere: https://codereview.qt-project.org/#/c/232286/6

**Comment by plfiorini** [comment]
See also:
* https://github.com/lirios/materialdecoration/issues/4
* https://bugreports.qt.io/browse/QTBUG-71340

**Comment by plfiorini** [comment]
As for decorations I will try to come up with a quick workaround but the long term solution is to abandon QtWayland client side decorations and do server side decorations or client side with QML.

**Comment by n3rdopolis** [comment]
worked around it with
```
# Workaround to install generated private headers.
QtWaylandVersion=$(ls -d ./include/QtWaylandClient/5.*| awk -F / '{print $4}')
cp ./src/client/qwayland*.h $INSTALLDIR/include/QtWaylandClient/$QtWaylandVersion/QtWaylandClient/private
cp ./src/client/wayland*.h $INSTALLDIR/include/QtWaylandClient/$QtWaylandVersion/QtWaylandClient/private
```
in my `qtwayland` build script...

**Comment by plfiorini** [comment]
fullscreen-shell is now in qtwayland dev: http://code.qt.io/cgit/qt/qtwayland.git/commit/?h=dev&id=7945996fe4360f12a96faad7718325b2a66774b4

**Comment by plfiorini** [comment]
@n3rdopolis Do you use qtwayland dev in your builds?

**Comment by n3rdopolis** [comment]
I use 5.12, not dev. Dev is 5.13 I think...

**Comment by plfiorini** [comment]
liri-wayland has been split in various pieces: fullscreen-shell is upstreamed and will be available with Qt 5.13, materialdecoration has its own repository but it will be deprecated soon, some code went into liri-shell and the QPA part is in liri-eglfs.
